week-2/Homework/extra/2-array-methods.js

```diff
@@ -5,31 +5,80 @@ Following are set of small functions that you need to create using loops and arr
 //1)Find the Smallest Number in an Array
 //Create a function that takes an array of numbers and return the smallest number in it
 
-function findSmallestNum(arr) {}
+function findSmallestNum(arr) {
 
-findSmallestNum([354, 15, 114, 2]); //Expected: 2
+    var smallestValue= Math.min.apply(Math,arr);
+    return smallestValue;
 
-findSmallestNum([-76, 1, -79, 1, 0]); //Expected: -79
+   }
+
+console.log(findSmallestNum([354, 15, 114, 2])); //Expected: 2
+
+console.log(findSmallestNum([-76, 1, -79, 1, 0])); //Expected: -79
 
 //2)Checking Even Numbers
 //Create a function that takes in an array and returns true if all its values are even, and false otherwise.
-function checkAllEven(arr) {}
+function checkAllEven(arr) {
+    for(var i=0;i<arr.length;i++){
+        if(arr[i]%2===0){
+            return true;
+        }
+        else{

```

Comment:
Think about what would happen if you called your function `checkAllEven` in the following way:
`checkEven([2, 3, 4, 6])`
According to your loop you check if a number is even then return true straight away - can you think why this might be a problem in this case ?